lib/core-integration/src/Test/Integration/Scenario/API/Byron/Migrations.hs

```diff
@@ -261,34 +272,36 @@ spec = describe "BYRON_MIGRATIONS" $ do
         addrs <- listAddresses @n ctx wNew
         let addr1 = (addrs !! 1) ^. #id
 
-        let payloadMigrate =
-                Json [json|
-                    { passphrase: #{fixturePassphrase}
-                    , addresses: [#{addr1}]
-                    }|]
-        rm <- request @[ApiTransaction n] ctx
-            (Link.migrateWallet @'Byron wOld)
-            Default
-            payloadMigrate
-        verify rm
-            [ expectResponseCode HTTP.status202
-            , expectField id ( (`shouldBe` 19) . length )
-            ]
-
-        -- Check that funds become available in the target wallet:
-        let expectedBalance = originalBalance - expectedFee - leftovers
-        eventually "wallet balance = expectedBalance" $ do
+        -- NOTE
+        -- The migration typically involves many transactions being sent one by
+        -- one. It may happen that one of these transaction is rolled back and
+        -- simply discarded entirely from mem pools. There's no retry mechanism
+        -- from the wallet _yet_, which means that such transactions must be
+        -- manually retried by clients.
+        --
+        -- This 'migrateWallet' function does exactly this, and will try to make
+        -- sure that rolledback functions are canceled and retried up until the
+        -- full migration is done.
+        liftIO $ migrateWallet ctx wOld [addr1]
+
+        -- Check that funds become available in the target wallet: Because
+        -- there's a bit of non-determinism in how the migration is really done,
+        -- we can expect the final balance with exactitude. Yet, we still expect
+        -- it to be not too far away from an ideal value.
+        let expectedMinBalance = originalBalance - 2 * expectedFee - leftovers
+        eventually "wallet balance ~ expectedBalance" $ do
             request @ApiWallet ctx
                 (Link.getWallet @'Shelley wNew)
                 Default
                 Empty >>= flip verify
                 [ expectField
                         (#balance . #getApiT . #available)
-                        ( `shouldBe` Quantity expectedBalance)
+                        (.> (Quantity expectedMinBalance))
                 , expectField
                         (#balance . #getApiT . #total)
-                        ( `shouldBe` Quantity expectedBalance)
+                        (.> (Quantity expectedMinBalance))
                 ]
+
         -- Analyze the target wallet UTxO distribution
```

Comment:
I get this from the `((`shouldBe` (Just 100)) . Map.lookup 10_000_000_000)` line below:
```
  src/Test/Integration/Scenario/API/Shelley/Migrations.hs:250:19:
  1) API Specifications, SHELLEY_MIGRATIONS, SHELLEY_MIGRATE_01_big_wallet -  migrate a big wallet requiring more than one tx
       While verifying (Status {statusCode = 200, statusMessage = "OK"},Right (ApiUtxoStatistics {total = Quantity {getQuantity = 10000094280000}, scale = ApiT {getApiT = Log10}, distribution = fromList [(10,0),(100,0),(1000,0),(10000,0),(100000,0),(1000000,84),(10000000,8),(100000000,0),(1000000000,0),(10000000000,0),(100000000000,100),(1000000000000,0),(10000000000000,0),(100000000000000,0),(1000000000000000,0),(10000000000000000,0),(45000000000000000,0)]}))
       expected: Just 100
        but got: Just 0

  To rerun use: --match "/API Specifications/SHELLEY_MIGRATIONS/SHELLEY_MIGRATE_01_big_wallet -  migrate a big wallet requiring more than one tx/"

Randomized with seed 151129959
```

Reply:
My bad, I did a wrong copy-paste thinking the assertions were the same for both shelley and byron but they aren't. That's been fixed shortly after but I assume you had the former branch :no_mouth:

Reply:
Seems to work on my computer...